### ISSUE TITLE
frog.cpython-310-darwin.so symbol not found on arm64 macos

### ISSUE THREAD

**Issue by EsamGhaleb** [proycon/python-frog#25]
Hello, 

I tried using the Python wrapper of frog, but I get the following error:

`anaconda3/envs/py31/lib/python3.10/site-packages/frog.cpython-310-darwin.so, 0x0002): symbol not found in flat namespace`

Is there a way to fix the error?

**Comment by EsamGhaleb** [comment]
I also get the following error when I run:  `python -c "import frog; frog.installdata()`

```
sh: wget: command not found
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "frog_wrapper.pyx", line 243, in frog.installdata
Exception: Installation of uctodata failed
```

Unexpected error. Is the library tested for Macbook M1?

**Comment by proycon** [comment]
The second error seems due to a missing `wget` on your system. Is it installed? (perhaps you need to do `brew install wget`, although I thought it shipped by default on macos)

> Unexpected error. Is the library tested for Macbook M1?

Yes, there are some M1 users and the CI pipeline also builds for it, though I myself don't have a mac to test.

The symbol not found error is more concerning. It wonder it if is due to an inconsistency on your system or due an actual problem in the Python wheel. Can you see if it reproduces without Anaconda, in a normal Python virtualenv? I'll also see if I can find a colleague with a mac to test it again.

**Comment by EsamGhaleb** [comment]
For the first error, indeed, the problem was related to the Anaconda distribution. Thanks a lot for pointing this out. I have fixed the issues. 

BTW, do you have trained models for Dutch word embeddings?

**Comment by jorisvanzundert** [comment]
@EsamGhaleb Can you report how you fixed this? @proycon: I'm running into the same weirdness in a normal .venv (Apple M1). I forced a recompile to see if that would yield some more information:

```
$pip install --force-reinstall --ignore-installed --no-binary :all: python-frog
DEPRECATION: --no-binary currently disables reading from the cache of locally built wheels. In the future --no-binary will not influence the wheel cache. pip 23.1 will enforce this behaviour change. A possible replacement is to use the --no-cache-dir option. You can use the flag --use-feature=no-binary-enable-wheel-cache to test the upcoming behaviour. Discussion can be found at https://github.com/pypa/pip/issues/11453
Collecting python-frog
  Downloading python-frog-0.6.6.tar.gz (98 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 98.4/98.4 kB 4.8 MB/s eta 0:00:00
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Collecting Cython
  Using cached Cython-3.0.0.tar.gz (2.7 MB)
  Preparing metadata (setup.py) ... done
Building wheels for collected packages: python-frog
  Building wheel for python-frog (pyproject.toml) ... error
  error: subprocess-exited-with-error
  
  × Building wheel for python-frog (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [1 lines of output]
      error: unknown file type '.pxd' (from 'frog_classes.pxd')
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for python-frog
Failed to build python-frog
ERROR: Could not build wheels for python-frog, which is required to install pyproject.toml-based projects
```

Any ideas? I can confirm Frog runs and functions fine from the command line.

**Comment by proycon** [comment]
@jorisvanzundert Sorry for the delay! (holiday period).  Forcing a recompilation only works if you already have Frog (the library and C++ headers), and all its dependencies, installed, though I'm a bit puzzled as to the error on `.pxd` you are getting. Neverthless, using the pre-compiled wheels in a normal virtualenv is the preferred route. Do you still get a symbol not found error when doing so?

**Comment by proycon** [comment]
I had a colleague with a mac try and he experienced the same problem with `python_frog-0.6.6-cp311-cp311-macosx_11_0_arm64.whl`. I'm going to build new wheels in the hope that helps...

```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ImportError: dlopen(/Users/dorianh/virtualenvs/frog/lib/python3.11/site-packages/frog.cpython-311-darwin.so, 0x0002): symbol not found in flat namespace '__ZN4TiCC10CL_Options6insertERKNSt3__112basic_stringIcNS1_11char_traitsIcEENS1_9allocatorIcEEEES9_'
```

**Comment by proycon** [comment]
Though the original issue is fixed. I haven't been able to fix the larger problem yet that the arm64 wheels on macos seem not to work anymore, intel wheels still work fine, and Linux works fine too. I wonder why this regressed after finally managing to get it right a while back. I'll continue investigating...

**Comment by proycon** [comment]
I just released python-frog 0.6.7 which should fix this issue the issue with the wheels for arm64 macos, as well as the compile issue reported by @jorisvanzundert . Closing this issue now.
